pkg/eigenState/stateMigrator/stateMigrations/202502031222_operatorSets/migration.go

```diff
@@ -0,0 +1,144 @@
+package _02502031222_operatorSets
+
+import (
+	"database/sql"
+	"github.com/Layr-Labs/sidecar/internal/config"
+	"github.com/Layr-Labs/sidecar/pkg/eigenState"
+	"github.com/Layr-Labs/sidecar/pkg/eigenState/stateManager"
+	"github.com/Layr-Labs/sidecar/pkg/eigenState/stateMigrator/helpers"
+	"github.com/Layr-Labs/sidecar/pkg/eigenState/stateMigrator/types"
+	"github.com/Layr-Labs/sidecar/pkg/storage"
+	"github.com/Layr-Labs/sidecar/pkg/utils"
+	"go.uber.org/zap"
+	"gorm.io/gorm"
+)
+
+type StateMigration struct {
+	db           *gorm.DB
+	logger       *zap.Logger
+	globalConfig *config.Config
+}
+
+func NewStateMigration(db *gorm.DB, l *zap.Logger, cfg *config.Config) types.IStateMigration {
+	return &StateMigration{
+		db:           db,
+		logger:       l,
+		globalConfig: cfg,
+	}
+}
+
+func (sm *StateMigration) GetMigrationName() string {
+	return "02502031222_operatorSets"
+}
+
+func (sm *StateMigration) MigrateState(currentBlockNumber uint64) ([][]byte, map[string][]interface{}, error) {
+	// This migration only applies to preprod and holesky. Mainnet will be released with all
+	// of these features (slashing, rewards v2.1) at once rather than iteratively like we did
+	// with preprod and holesky
+	if sm.globalConfig.Chain == config.Chain_Mainnet {
+		sm.logger.Sugar().Infof("No migration needed for mainnet")
+		return nil, nil, nil
+	}
+
+	stateMan := stateManager.NewEigenStateManager(nil, sm.logger, sm.db)
+	if err := eigenState.LoadEigenStateModels(stateMan, sm.db, sm.logger, sm.globalConfig); err != nil {
+		sm.logger.Sugar().Errorw("Failed to load eigen state models for migration", zap.Error(err))
+		return nil, nil, err
+	}
+
+	contracts := sm.globalConfig.GetContractsMapForChain()
+
+	query := `
+		select
+			tl.*
+		from transaction_logs as tl
+		where
+			tl.address = @allocationManagerAddress
+			and event_name in @eventNames
+			and block_number < @currentBlockNumber
```

Comment:
Just want to double check that we're indeed capturing `currentBlockNumber` as part of the regular ingestion.